### PR TITLE
std.math: Begin deprecation of accidentally leaked AliasSeq

### PR DESCRIPTION
**Pull request by ibuclaw**
~No, there isn't a better way of doing this.  Complements to @schveiguy for an initial version of this PR.~

Rebased and updated following dmd fix.

### CODE REVIEW

**Comment by MoonlightSentinel** [comment]
That workaround is really ugly...
 
The weird part is that this bug doesn't apply to deprecated function imports.

**Comment by dnadlinger** [comment]
Maybe `std.meta.AliasSeq was unintentionally available from std.math and will be removed after …. Please import std.meta instead.` or something that makes clear where the symbol came from? If I saw this as an user not too familiar with metaprogramming in D, I would try to look up docs for std.math.AliasSeq to figure out what was going on.

**Comment by schveiguy** [comment]
The worst part about this is that the deprecation message will be printed for the mixin line, not the usage. However, one possible saving grace is that the full instantiation is printed, so you might be able to deduce where it's coming from based on that.

File: std/math.d [hunk 1]
```diff
@@ -133,8 +133,15 @@ static import core.stdc.fenv;
 import std.traits :  CommonType, isFloatingPoint, isIntegral, isNumeric,
     isSigned, isUnsigned, Largest, Unqual;
 
+// @@@DEPRECATED_2.102@@@
 // Note: Exposed accidentally, should be deprecated / removed
-public import std.meta : AliasSeq;
+private template DeprecatedAliasSeq(V...)
```
Comment (schveiguy) [inline std/math.d:138]: I think you need to make this public, no?

Comment (ibuclaw) [reply std/math.d:138]: No, otherwise we'd have to deprecated the DeprecatedAliasSeq too.

Comment (schveiguy) [reply std/math.d:138]: Can you publicly alias a private symbol? I didn't think that was possible.

Comment (ibuclaw) [reply std/math.d:138]: It works for me.

Comment (schveiguy) [reply std/math.d:138]: OK, you are right, I tested locally and it works.

**Comment by MoonlightSentinel** [comment]
Please don't merge this yet,  I think i have a fix for the missing deprecation.

File: std/math.d [hunk 2]
```diff
@@ -133,8 +133,15 @@ static import core.stdc.fenv;
...
+{
+    deprecated("std.meta.AliasSeq was unintentionally available from std.math and will be removed after 2.102. Please import std.meta instead")
```
Comment (schveiguy) [inline std/math.d:140]: You need to break this line in half, style check complains.

**Comment by schveiguy** [comment]
Should we close this now that https://github.com/dlang/dmd/pull/11158 is merged?

**Comment by MoonlightSentinel** [comment]
Not really, just add a plain deprecation message?

**Comment by schveiguy** [comment]
Right, maybe adjust this to properly deprecate the import.

**Comment by 12345swordy** [comment]
@ibuclaw ping

**Comment by ibuclaw** [comment]
Rebased.

**Pull request merged**